src/template/L1PortalExecuteL2Call.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {VmSafe} from "forge-std/Vm.sol";
+import {stdToml} from "forge-std/StdToml.sol";
+
+import {MultisigTaskPrinter} from "../libraries/MultisigTaskPrinter.sol";
+import {Action} from "../libraries/MultisigTypes.sol";
+import {SimpleTaskBase} from "../tasks/types/SimpleTaskBase.sol";
+
+/// @notice Interface for the OptimismPortal2 contract on L1.
+interface IOptimismPortal2 {
+    function depositTransaction(address _to, uint256 _value, uint64 _gasLimit, bool _isCreation, bytes memory _data)
+        external
+        payable;
+}
+
+/// @notice Template to execute an L2 call via the L1 Optimism Portal from a nested L1 Safe.
+/// Sends an L2 transaction using OptimismPortal.depositTransaction with config-driven params.
+contract L1PortalExecuteL2Call is SimpleTaskBase {
```

Comment:
I was thinking this _could_ inherit from `L2TaskBase` instead of SimpleTaskBase if you anticipate bundling many L1Portal calls for many chains in the one task. The OptimismPortal added doesn't need to be added to the config.toml file then. It'll be available via something like: 

```
        OptimismPortal optimismPortal =
            OptimismPortal(superchainAddrRegistry.getAddress("OptimismPortalProxy", chains[0].chainId));
```

Reply:
I do like the idea of not needing to specify the portal in the config file, removes a source of error. Having a single target / array length of one is ok, if I understood the question correctly

Reply:
Yea I agree with @mds1. Let's change this to inherit from `L2TaskBase`. Having an array of 1 is totally fine, we do this frequently.